extensions/security-jpa/runtime/src/main/java/io/quarkus/security/jpa/JpaRecorder.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.security.jpa;
+
+import java.security.Security;
+
+import org.wildfly.security.password.WildFlyElytronPasswordProvider;
+
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class JpaRecorder {
+
+    public void registerSecurityProvider() {
+        Security.addProvider(new WildFlyElytronPasswordProvider());
```

Comment:
This is also registered by io.quarkus.elytron.security.runtime.ElytronRecorder#registerPasswordProvider, I don't think we want it twice (not sure if it matters)? If registering it twice causes problems we probably need some build items to manage the provider registration.

Reply:
IIRC it is conditionally added only if you have a realm, which is not the case here.

Reply:
So it's not conditional anymore, but I'd have to make my extension depend on `quarkus-security-elytron` to get it. Do you think that's fine?

Reply:
I removed my recorder and depend on that extension. I guess it should be fine.

Reply:
We might consider switching this to be ServiceLoader-based.  All of the Elytron providers are available via `ServiceLoader.load(Provider.class)` normally; we could have a startup step where we register all providers on the class path at boot.

The JDK has already starting moving to this behavior in fact, and moving away from requiring global provider registration in some cases [1].  So really, we probably should look at enhancing the Elytron APIs to use this approach as well.